### DISCUSSION TITLE
Comparing a ref to null?

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#2401] [Language Ideas] [answered]
I'm being dumb I'm sure but I have a method that returns a ref to an unmanaged struct.

```cs
ref var block_ref = ref SomeMethod();
```

I want to now write:

```cs
if (block_ref == null)
   ...
```

But I can't get a clean compile no matter what I try...

**Comment by CyrusNajmabadi** [comment]
@Korporal You might want to consider gitter.im/dotnet/csharplang for quick questions like this.  For the above, could you explain how you would get null in the first place in block_ref?

**Comment by tannergooding** [comment, marked as answer]
You can only get a `null ref` (that is a ref which is null; rather than a reference to a null) via unsafe means today. For example, using `Unsafe.AsRef<T>(null)` or via interop code involving ref returns.

Likewise, you need to use the `Unsafe` class to check if you have a `null ref`.

**Comment by The-Futurist** [comment]
@CyrusNajmabadi - OK You're right, I've not been to gitter and I guess I should take a look.

**Comment by The-Futurist** [comment]
> You can only get a `null ref` (that is a ref which is null; rather than a reference to a null) via unsafe means today. For example, using `Unsafe.AsRef<T>(null)` or via interop code involving ref returns.
> 
> Likewise, you need to use the `Unsafe` class to check if you have a `null ref`.

@tannergooding 

Yes and that's what this is, under the hood the method attempts to get some memory (from outside the AppDomain but inside the process address space) if it is not able to then ordinarily we'd return a null pointer (for example if this in C).

**Comment by The-Futurist** [comment]
I've moved to gitter...
